src/TypeSystem/Scope/TernaryBranchScope/TernaryBranchScope.php

```diff
@@ -26,13 +26,12 @@
 use PackageFactory\ComponentEngine\Parser\Ast\TypeReferenceNode;
 use PackageFactory\ComponentEngine\TypeSystem\Narrower\NarrowedTypes;
 use PackageFactory\ComponentEngine\TypeSystem\Narrower\Expression\ExpressionTypeNarrower;
-use PackageFactory\ComponentEngine\TypeSystem\Narrower\TypeNarrowerContext;
 use PackageFactory\ComponentEngine\TypeSystem\ScopeInterface;
 use PackageFactory\ComponentEngine\TypeSystem\TypeInterface;
 
 final class TernaryBranchScope implements ScopeInterface
 {
-    private function __construct(
+    public function __construct(
```

Comment:
Why that?

Reply:
Because one should use the static factories - but I guess this will make unit testing harder … for now fine imo? I like having private constructors at times …

Reply:
But you've made it `public` :monocle_face:

Reply:
Misread the diff 😂 its the 4 day in berlin 😅😂